src/compute/src/render/join/linear_join.rs

```diff
@@ -466,43 +472,45 @@ where
 {
     match (prev_keyed, next_input) {
         (
-            SpecializedArrangementImport::Bytes9Row(prev_key_types, prev_keyed),
-            SpecializedArrangement::Bytes9Row(next_key_types, next_input),
-        ) => {
-            assert!(prev_key_types
-                .iter()
-                .zip(next_key_types.iter())
-                .all(|(c1, c2)| c1.scalar_type == c2.scalar_type));
-            differential_join_inner(
-                join_spec,
-                prev_keyed,
-                next_input,
-                Some(prev_key_types),
-                None,
-                None,
-                closure,
-            )
-        }
+            SpecializedArrangementImport::RowUnit(prev_keyed),
+            SpecializedArrangement::RowUnit(next_input),
+        ) => differential_join_inner(
+            join_spec,
+            prev_keyed,
+            next_input,
+            None,
+            Some(vec![]),
+            Some(vec![]),
+            closure,
+        ),
         (
-            SpecializedArrangementImport::RowRow(prev_keyed),
+            SpecializedArrangementImport::RowUnit(prev_keyed),
             SpecializedArrangement::RowRow(next_input),
-        ) => differential_join_inner(join_spec, prev_keyed, next_input, None, None, None, closure),
-        (
-            SpecializedArrangementImport::Bytes9Row(key_types, _),
-            SpecializedArrangement::RowRow(_),
-        ) => panic!(
-            "Invalid combination of type specializations: key types differ! \
-            Bytes9 trace vs. Row local, key_types: {:?}",
-            key_types
+        ) => differential_join_inner(
+            join_spec,
+            prev_keyed,
+            next_input,
+            None,
+            Some(vec![]),
+            None,
+            closure,
```

Comment:
Uncovered

Reply:
Same as above; please reopen this conversation if not true.